packages/toolpad-app/src/server/data.ts

```diff
@@ -531,16 +531,26 @@ export async function loadRenderTree(appId: string, version: VersionOrPreview =
 
 export async function duplicateApp(id: string, name: string): Promise<AppMeta> {
   const dom = await loadPreviewDom(id);
-  const duplicateCount = await prismaClient.app.count({
-    where: { name: { startsWith: `${name} (copy`, endsWith: ')' } },
-  });
-  const duplicateName =
-    duplicateCount === 0 ? `${name} (copy)` : `${name} (copy ${duplicateCount + 1})`;
-  const newApp = await createApp(duplicateName, {
-    from: {
-      kind: 'dom',
-      dom,
-    },
-  });
-  return newApp;
+  try {
+    const newApp = await createApp(name, {
+      from: {
+        kind: 'dom',
+        dom,
+      },
+    });
+    return newApp;
+  } catch (error) {
+    const duplicateCount = await prismaClient.app.count({
```

Comment:
This needs to only run when the error is the unique constraint of the name column. All other errors need to be rethrown. We wouldn't want this to run when something like a network error has happened